locator/tablets.cc

```diff
@@ -220,6 +223,25 @@ dht::token_range tablet_map::get_token_range(tablet_id id) const {
     }
 }
 
+seastar::future<std::vector<token>> tablet_map::get_flat_token_ranges() const {
+    std::vector<token> tokens;
+    tokens.reserve(tablet_count() + 1u);
+
+    for (auto id : tablet_ids()) {
+        auto range = get_token_range(id);
```

Comment:
Each tablet has exactly one token. Use `tablet_map::get_last_token()` to obtain this.

Reply:
Explanation: in both the ring and tablets, there are individual tokens, which form a ring. In the ring, the range of a given vnode/tablet extends from the last token of the previous vnode/tablet to that of the current one. This is implicit, so you don't need to special handle the first range.